### PR TITLE
refactor: change from pagination to scroll on badgedex

### PR DESCRIPTION
**Pull request by Darguima**
Added a fallback image to the Badges, when the url one isn't available (on the video all them are the fallback)
Added Empty badges message.
Added Infinite Scroll (with a Go To Top button)

https://github.com/cesium/seium.org/assets/49988070/d5feeca7-43f1-4e61-860e-bacc2f4d7d9b

[EDIT] - changed video to also have the sliding effect on the button

### CODE REVIEW

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]
Can you use this badge image instead?
image

**Comment by Darguima** [comment]
I already edited, but the 404 have a buggy shadow

image

**Comment by Darguima** [comment]
Fixed

image

File: components/Badge/index.tsx [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
 
-export default function Badge({ name, id, avatar, tokens, owned }) {
   return (
     <Link
       href={`/badge/${id}`}
       className={`h-full w-full ${owned ? "opacity-100" : "opacity-30"}`}
     >
       <div>
-        <img src={avatar} alt={name}></img>
+        <img src={avatar} onError={imageOnError} alt="Badge Image"></img>
```
Comment (ruilopesm) [inline components/Badge/index.tsx:23]:
```suggestion
        <img src={avatar} onError={imageOnError} alt={name}></img>
```
For accessibility reasons.

**Review by ruilopesm — COMMENTED** [review]
Great work 🙌🏼
In fact, this is not an infinite scroll, as the content is not being loaded as I scroll down. However, I feel that approach is over complicated and not necessary here. Anyway, I think we should implement a skeleton rendering for when a badge is rendering - can you handle that? 💪🏼

**Comment by Darguima** [comment]
> Great work 🙌🏼 In fact, this is not an infinite scroll, as the content is not being loaded as I scroll down. However, I feel that approach is over complicated and not necessary here. Anyway, I think we should implement a skeleton rendering for when a badge is rendering - can you handle that? 💪🏼

Done. Any suggestions or any already existing color schema on the project that I can use here for the skeleton?

Also, I suggest you to try because the video is a little bit confusing on the transition due quick access to the image.


https://github.com/cesium/seium.org/assets/49988070/a564664a-2ab2-40b9-9a9e-406a8d787a4f

**Comment by Darguima** [comment]
I fixed that delay showing the fallback image after the skeleton. Also fixed the infinite loop fallback.

**Pull request merged**